main/test/ca/uwaterloo/flix/library/TestFile.flix

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2021 Nina Andrup Pedersen
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace TestFile {
+
+    use File.{exists, isDirectory, isRegularFile, isReadable, isSymbolicLink,
+        isWritable, isExecutable};
+
+    /////////////////////////////////////////////////////////////////////////////
+    // exists                                                                  //
+    /////////////////////////////////////////////////////////////////////////////
+
+    @test
+    def testExists01(): Bool & Impure =
+        exists("./README.md") == Ok(true)
+
+    @test
+    def testExists02(): Bool & Impure =
+        exists("./fakeREADME.md") == Ok(false)
+
+    /////////////////////////////////////////////////////////////////////////////
+    // isDirectory                                                             //
+    /////////////////////////////////////////////////////////////////////////////
+
+    @test
+    def testIsDirectory01(): Bool & Impure =
+        isDirectory("./main/test/ca/uwaterloo/flix/library") == Ok(true)
+
+    @test
+    def testIsDirectory02(): Bool & Impure =
+        isDirectory("./README.md") == Ok(false)
+
+    /////////////////////////////////////////////////////////////////////////////
+    // isRegularFile                                                           //
+    /////////////////////////////////////////////////////////////////////////////
+
+    @test
+    def testIsRegularFile01(): Bool & Impure =
+        isRegularFile("./main/test/ca/uwaterloo/flix/library") == Ok(false)
+
+    @test
+    def testIsRegularFile02(): Bool & Impure =
+        isRegularFile("./README.md") == Ok(true)
+
+    /////////////////////////////////////////////////////////////////////////////
+    // isReadable                                                              //
+    /////////////////////////////////////////////////////////////////////////////
+
+    @test
+    def testIsReadable01(): Bool & Impure =
+        isReadable("./README.md") == Ok(true)
+
+    @test
+    def testIsReadable02(): Bool & Impure =
+        isReadable("./fakeREADME.md") == Ok(false)
+
+    /////////////////////////////////////////////////////////////////////////////
+    // isSymbolicLink                                                          //
+    /////////////////////////////////////////////////////////////////////////////
+
+    // TODO: find symbolic link to test
```

Comment:
I could not find a file to test for symbolic link

Reply:
I think for many of them one test is fine.

Reply:
Just test what you can and leave the rest. We don't need to test both outcomes of each function. One outcome is enough.